### PR TITLE
POC: enable to train at the double precision

### PR DESCRIPTION
**Pull request by raimis**

### CODE REVIEW

File: torchmdnet/scripts/train.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def get_args():
     parser.add_argument('--ema-alpha-neg-dy', type=float, default=1.0, help='The amount of influence of new losses on the exponential moving average of dy')
     parser.add_argument('--ngpus', type=int, default=-1, help='Number of GPUs, -1 use all available. Use CUDA_VISIBLE_DEVICES=1, to decide gpus')
     parser.add_argument('--num-nodes', type=int, default=1, help='Number of nodes')
-    parser.add_argument('--precision', type=int, default=32, choices=[16, 32], help='Floating point precision')
+    parser.add_argument('--precision', type=int, default=32, choices=[16, 32, 64], help='Floating point precision')
```
Comment (RaulPPelaez) [inline torchmdnet/scripts/train.py:40]: oh wow I totally missed this argument when I implemented #182

File: torchmdnet/module.py [hunk 2]
```diff
@@ -127,8 +127,9 @@ def step(self, batch, loss_fn, stage):
             if batch.y.ndim == 1:
                 batch.y = batch.y.unsqueeze(1)
 
-            # y loss
-            loss_y = loss_fn(y, batch.y)
+            # y
+            y_dtype = {16: torch.float16, 32: torch.float32, 64: torch.float64}[self.hparams.precision]
+            loss_y = loss_fn(y, batch.y.to(y_dtype))
```
Comment (RaulPPelaez) [inline torchmdnet/module.py:132]: How come you need this here but not a few lines above for neg_dy?

File: torchmdnet/datasets/ace.py [hunk 3]
```diff
@@ -180,6 +182,11 @@ def sample_iter(self, mol_ids=False):
                 fq = pt.tensor(mol["formal_charges"], dtype=pt.long)
                 q = fq.sum()
 
+                # Keep molecules with specific elements
+                if self.atomic_numbers:
+                    if not set(z.numpy()).issubset(self.atomic_numbers):
+                        continue
```
Comment (RaulPPelaez) [inline torchmdnet/datasets/ace.py:188]: This got mixed from #206, right?

File: torchmdnet/datasets/ace.py [hunk 4]
```diff
@@ -309,9 +317,7 @@ def get(self, idx):
         atoms = slice(self.idx_mm[idx], self.idx_mm[idx + 1])
         z = pt.tensor(self.z_mm[atoms], dtype=pt.long)
         pos = pt.tensor(self.pos_mm[atoms], dtype=pt.float32)
-        y = pt.tensor(self.y_mm[idx], dtype=pt.float32).view(
-            1, 1
-        )  # It would be better to use float64, but the trainer complaints
+        y = pt.tensor(self.y_mm[idx], dtype=pt.float64).view(1, 1)
```
Comment (RaulPPelaez) [inline torchmdnet/datasets/ace.py:320]: I would pass dtype as an argument to Ace here and store everything in the correct type. I do not see why store pos in float32 and y in float64.

**Comment by raimis** [comment]
Obsolete

**Pull request closed without merging**